sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -1479,4 +1481,20 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
         |FROM (SELECT '{"f1": "value1", "f2": 12}' json, 'hello' as str) test
       """.stripMargin), Row("value1", "12", 3.14, "hello"))
   }
+
+  // This test case is to verify a bug when making a new instance of LogicalRDD.
+  test ("SPARK-11633: HiveContext throws TreeNode Exception : Failed to Copy Node") {
```

Comment:
some suggestion about the test:

* do not put tests in hive module if it's not hive related. Actually we can reproduce this bug by setting the case sensitive config to false and put the test in `DataFrameSuite`.
* do not create new classes unless we have to. For this case we can create RDD[Row] and specify the schema for it.
* clean up temp table at the end of test, or only use the `DataFrame` API

here is an example and hope you can follow it in the future :)
```
withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false") {
  val rdd = sparkContext.makeRDD(Seq(Row(1, 3), Row(2, 1)))
  val df = sqlContext.createDataFrame(
    rdd,
    new StructType().add("f1", IntegerType).add("f2", IntegerType),
    needsConversion = false).select($"F1", $"f2".as("f2"))
  val df1 = df.as("a")
  val df2 = df.as("b")
  checkAnswer(df1.join(df2, $"a.f2" === $"b.f2"), Row(2) :: Row(1) :: Nil)
}
```